vpd-manager/src/manager.cpp

```diff
@@ -640,4 +645,42 @@ void Manager::performVpdRecollection()
         m_worker->performVpdRecollection();
     }
 }
+
+int Manager::performVpdSanityCheck(const types::Path i_vpdPath)
+{
+    try
+    {
+        if (i_vpdPath.empty())
+        {
+            throw std::runtime_error("Given FRU path is empty");
+        }
+
+        nlohmann::json l_sysCfgJsonObj{};
+
+        if (m_worker.get() != nullptr)
+        {
+            l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
+        }
+
+        std::shared_ptr<Parser> l_parserObj =
+            std::make_shared<Parser>(i_vpdPath, l_sysCfgJsonObj);
+
+        std::shared_ptr<vpd::ParserInterface> l_vpdParserInstance =
```

Comment:
This is different from the flow that we discussed?
Parser class should expose an API "performVpdSanityCheck" where all the checks against JSON needs to be done?
Based on that implementation should proceed and create parser instance and perform sanity check?